sonar-ruby-plugin/src/test/java/org/sonarsource/ruby/converter/visitor/StringLiteralVisitorTest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.ruby.converter.visitor;
+
+import java.util.Arrays;
+import org.junit.Test;
+import org.sonarsource.ruby.converter.AbstractRubyConverterTest;
+import org.sonarsource.slang.api.NativeTree;
+import org.sonarsource.slang.api.StringLiteralTree;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonarsource.slang.testing.TreeAssert.assertTree;
+
+public class StringLiteralVisitorTest extends AbstractRubyConverterTest {
+
+  @Test
+  public void plain_string_literal() {
+    StringLiteralTree tree = (StringLiteralTree) rubyStatement("'foo'");
+    assertTree(tree).isStringLiteral("foo");
+  }
+
+  @Test
+  public void file_macro() {
+    NativeTree tree = (NativeTree) rubyStatement("__FILE__");
+    assertThat(((NativeTree) tree.children().get(0)).nativeKind()).isEqualTo(nativeKind("(SonarRuby analysis)"));
```

Comment:
Is it ok to leave native kind as `"(SonarRuby analysis)"` ? we could maybe set it to something more explicit such as `"FILE_MACRO"`.
Not sure what's best... you can leave it as is if you prefer it like that :)

Reply:
I would keep this, should be obvious what this means

Reply:
Ok. Fine by me :)